pkg/controllers/node/termination/terminator/terminator.go

```diff
@@ -72,23 +77,29 @@ func (t *Terminator) Taint(ctx context.Context, node *v1.Node) error {
 		if err := t.kubeClient.Patch(ctx, node, client.StrategicMergeFrom(stored)); err != nil {
 			return err
 		}
-		logging.FromContext(ctx).Infof("tainted node")
+		logging.FromContext(ctx).With("taint.Key", taint.Key).With("taint.Effect", taint.Effect).With("taint.Value", taint.Value).Infof("tainted node")
 	}
 	return nil
 }
 
 // Drain evicts pods from the node and returns true when all pods are evicted
 // https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown
-func (t *Terminator) Drain(ctx context.Context, node *v1.Node) error {
+func (t *Terminator) Drain(ctx context.Context, node *v1.Node, nodeGracePeriodExpirationTime *time.Time) error {
 	pods, err := nodeutil.GetPods(ctx, t.kubeClient, node)
 	if err != nil {
 		return fmt.Errorf("listing pods on node, %w", err)
 	}
+
 	// evictablePods are pods that aren't yet terminating are eligible to have the eviction API called against them
 	evictablePods := lo.Filter(pods, func(p *v1.Pod, _ int) bool { return podutil.IsEvictable(p) })
+
+	if err := t.DeleteExpiringPods(ctx, evictablePods, nodeGracePeriodExpirationTime); err != nil {
```

Comment:
Since `do-not-disrupt` pods aren't considered evictable, they're no longer included in this call. I think we still want to include them here so they get their full `terminationGracePeriodSeconds` right (or at least the node's `terminationGracePeriod`? Also, should probably add some tests around this behavior, ensuring that `do-not-disrupt` pods are only deleted at expiration time - pod.terminationGracePeriodSeconds.

Reply:
I opted to use the evictable pods here to avoid deleting daemonset, mirror, etc pods. I still think it's reasonable to avoid evicting those. Is there another existing filter I could use to allow deletion of the do-not-disrupt pods but avoid the others? Does that make sense?

Reply:
Yep, that makes sense. There's no existing filter and I do think it makes sense to classify `do-not-disrupt` pods as "non-evictable" since we won't remove them using the eviction API. Maybe we introduce a new filter, something along the lines of `IsRemovable`? This would be the same as `IsEvictable` except it wouldn't filter `do-not-disrupt` pods.

Reply:
What do you think about filtering here with `IsReschedulable`? That covers both daemonsets and static pods, and will still delete pods with the do-not-disrupt annotation or blocking PDBs.

"If the pod can be rescheduled, we delete it preemptively to ensure it gets a full `terminationGracePeriodSeconds` of graceful termination time."

Reply:
I've switched this to use `IsReschedulable` instead, which I think meets the desired needs well.